pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,18 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
-    String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
-    if (timeout != null) {
+    long result = 0L;
+    if (PGProperty.LOGIN_TIMEOUT.isPresent(props)) {
+      String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = (long) (Float.parseFloat(timeout) * 1000);
       } catch (NumberFormatException e) {
         LOGGER.log(Level.WARNING, "Couldn't parse loginTimeout value: {0}", timeout);
       }
+    } else {
+      result = (long) DriverManager.getLoginTimeout() * 1000;
```

Comment:
Do you mean to use the `DriverManager.getLoginTimeout` only in case of option not being present in the props?
I think it makes sense to revert to default case in case of "parse failure" as well.